### PR TITLE
Week-3

### PR DESCRIPTION
**Pull request by EliaYazdi**

### CODE REVIEW

File: Week3/homework/step2-6.js [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      for (let l = 0; l < arr[i][j].length; l++) {
+
+        flattenArr3.push(arr[i][j][l]);
+      }
+    }
+
+  }
+  return flattenArr3;
```
Comment (grod220) [inline Week3/homework/step2-6.js:41]: This works, but you can also use the `.flat()` method :)

File: Week3/homework/step2-7.js [hunk 2]
```diff
@@ -21,3 +25,7 @@ f2(y);
 console.log(y);
 
 // Add your explanation as a comment here
+
+//the first function just changes the integer but we console.log the x which is 9.
+//the second function takes the value 'y' which is an object. It takes the value of x but then it updates the x
+//to become 10 and indeed it updated the 'y'to become equal to 10 as well.
```
Comment (grod220) [inline Week3/homework/step2-7.js:31]: I think you get it (?). Another example: 
```
let a = 4
let b = a
a += 1
What's the value of b?
```
vs objects...
```
let a = { name: 'gabe' }
b = a
a.country = 'us'
what's the value of b?
```
Do you see the difference?

File: Week3/homework/step3-bonus.js [hunk 3]
```diff
@@ -3,12 +3,12 @@
 const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 
 function makeUnique(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return Array.from(new Set(arr));
```
Comment (grod220) [inline Week3/homework/step3-bonus.js:6]: Nice 👍

File: Week3/homework/step3.js [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
 
 function createBase(base) {
-  // Replace this comment and the next line with your code
-  console.log(base);
+
+  const addNum = function (num) {
+    return base + num;
+  };
+  return addNum;
 }
```
Comment (grod220) [inline Week3/homework/step3.js:9]: Nice 👍

File: Week3/homework/step2-6.js (hunk 1 above)
Comment (EliaYazdi) [reply Week3/homework/step2-6.js:41]: @grod220 yeah , and it's much easier with .flat().

**Review by grod220 — APPROVED** [review]

**Pull request merged**